UnitTests/Generic/DebouncedEventArgsGenericTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿// SPDX-FileCopyrightText: 2021 Frans van Dorsselaer
+//
+// SPDX-License-Identifier: MIT
+//
+// SPDX-FileContributor: Alain van den Berg
+
+using System.Collections;
+
+namespace UnitTests.Generic;
+
+[TestClass]
+[TestCategory("Production")]
+public class DebouncedEventArgsGenericTests
+{
+    sealed class MockEnumerator(MockReadOnlyList List) : IEnumerator<int>
+    {
+        int Pos = -1;
+
+        public int Current => List[Pos];
+
+        object IEnumerator.Current => Current;
+
+        public void Dispose() { }
+
+        public bool MoveNext()
+        {
+            if (Pos >= List.Count || (Pos == -1 && List.Count == 0))
+            {
+                return false;
+            }
+            Pos++;
+            return true;
+        }
+
+        public void Reset()
+        {
+            Pos = -1;
+        }
+    }
+
+    sealed class MockReadOnlyList(int _Count) : IReadOnlyList<int>
+    {
+        public int this[int index]
+        {
+            get
+            {
+                if (index < 0 || index >= Count)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index));
+                }
+                return index;
+            }
+        }
+
+        public int Count => _Count;
+
+        IEnumerator IEnumerable.GetEnumerator() => new MockEnumerator(this);
+
+        public IEnumerator<int> GetEnumerator() => new MockEnumerator(this);
+    }
+
+    public static IEnumerable<object[]> ValidCounts
+    {
+        get
+        {
+            yield return new object[] { 1L, new MockReadOnlyList(0) };
```

Comment:
The documentation seems to say that an empty `triggerData` would throw an `ArgumentOutOfRangeException`?

Reply:
Fixed the documentation.